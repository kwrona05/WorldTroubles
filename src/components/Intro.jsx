import React, { useState, useEffect } from "react";

const Intro = ({ onFinish }) => {
  const sentences = [
    "Hunger, addictions, depression, wars, natural disasters, poverty...",
    "Almost all of us have heard about these problems, but are we aware of the biggest problem of the 21st century?",
    "Lack of information.",
    "Social media and technology are incredibly absorbing nowadays. As a result, we don't pay as much attention to the world around us.",
    "This application aims to raise people's awareness of the world around them.",
  ];

  const [typingSpeed] = useState(70);
  const [sentenceIndex, setSentenceIndex] = useState(0);
  const [charIndex, setCharIndex] = useState(0);
  const [currentText, setCurrentText] = useState("");
  const [isDeleting, setIsDeleting] = useState(false);

  useEffect(() => {
    console.log("Intro is rendering...");
    const handleTyping = () => {
      if (charIndex < sentences[sentenceIndex].length) {
        setCurrentText((prev) => prev + sentences[sentenceIndex][charIndex]);
        setCharIndex((prev) => prev + 1);
      } else {
        setTimeout(() => {
          setCurrentText("");
          setCharIndex(0);
          setSentenceIndex((prev) => (prev + 1) % sentences.length);
          if (sentenceIndex === sentences.length - 1) {
            if (onFinish) onFinish();
          }
        }, 500);
      }
    };

    const timeout = setTimeout(handleTyping, typingSpeed);
    return () => clearTimeout(timeout);
  }, [charIndex, isDeleting, sentenceIndex, sentences, typingSpeed, onFinish]);

  return (
    <div className="typing-div">
      <h2 data-testid="intro-header">
        {currentText}
        <span className="cursor" data-testid="cursor">
          |
        </span>
      </h2>
    </div>
  );
};

export default Intro;
