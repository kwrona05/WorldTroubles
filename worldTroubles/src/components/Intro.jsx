import { useEffect, useState, useRef } from "react";

const Intro = () => {
  const items = [
    "Hunger, addictions, depression, wars, natural disasters, poverty...",
    "Almost all of us have heard about these problems, but are we aware of the biggest problem of the 21st century?",
    "Lack of information",
    "Social media and technology are incredibly absorbing nowadays. As a result, we don't pay as much attention to the world around us.",
    "This application aims to raise people's awareness of the world around them.",
  ];

  const [currentIndex, setCurrentIndex] = useState(0);

  useEffect(() => {
    const interval = setInterval(() => {
      setCurrentIndex((prevIndex) => (prevIndex + 1) % items.length);
    }, 5000);

    return () => clearInterval(interval);
  }, [items.length]);

  return (
    <div className="detail-container">
      <h1 style={{ textAlign: "center" }}>{items[currentIndex]}</h1>
    </div>
  );
};

export default Intro;
