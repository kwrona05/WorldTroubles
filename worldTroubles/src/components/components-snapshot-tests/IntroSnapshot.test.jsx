import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Intro from "../Intro";
import { MemoryRouter } from "react-router-dom";

test("Matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Intro />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
