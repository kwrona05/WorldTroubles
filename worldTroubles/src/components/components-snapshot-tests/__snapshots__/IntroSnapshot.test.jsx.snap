// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="typing-div"
  >
    <h2
      data-testid="intro-header"
    >
      <span
        class="cursor"
        data-testid="cursor"
      >
        |
      </span>
    </h2>
  </div>
</DocumentFragment>
`;
